### ISSUE TITLE
memory usage  and CPU load.

### ISSUE THREAD

**Issue by Dylan404** [arkayenro/arkinventory#1388]
please check the memory usage and the CPU load. It's way too high. Patch 9.0.1.

**Comment by arkayenro** [comment]
nope, its uses tonnes of memory, and loads of CPU to process al the data.

if you can find something that is using an absurd amount of CPU i'd be happy to try an refactor the code to make it better, but the tradeoff will typically be using more memory to cache the calculation results

unless youre having a gameplay issue i dont really care how much memory it uses

**Comment by arkayenro** [comment]
that shouldnt be consistent though, it should build the data (which is cpu intensive) then use the cached results

what were you doing at the time?  do you have window sort set to immediate?

**Comment by ciphersimian** [comment]
I've got 100 addons and I've been consistently having problems with performance since the pre-patch.  I finally got fed up with it and gave up on addon updates to resolve the issue and downloaded Addon Usage.  I'm experiencing constant hiccups every 5s or so with much, much worse spikes of lag every 30m-1h.  For profiling I just used a very repeatable mount run in Vortex Pinnacle where I turned on the continuous CPU monitoring and hit reset outside the instance.  I then zoned in and cleared up to Altarius then jumped off to get ported back to entrance and zoned out.

Looking at the results, ArkInventory blew every other addon I have out of the water on CPU usage (even TSM, the most bloated piece of garbage ever created!)

The thing is, during this run I never even opened my bags.  I wouldn't expect any CPU usage at all from ArkInventory if I'm not even accessing my bags.

I did not have these problems before the pre-patch.  I believe something changed in an API you're calling or events that you're receiving that has absolutely tanked ArkInventory's performance.

I personally feel that this is the best bag addon out there and I make heavy use of the custom sorting and rules functionality.  I really don't want to have to find a replacement (because trust me I've tried all the others) but this performance is unacceptable.

**Comment by ciphersimian** [comment]
Here's the output from my 100 addon run:
https://imgur.com/a/MI6SWeZ

And here's a run with just ArkInventory after deleting the entirety of my WTF and AddOn folders, downloading a fresh copy of 3.09.43 and installing nothing but that and AddonUsage:
https://imgur.com/a/nQiliah

**Comment by arkayenro** [comment]
> The thing is, during this run I never even opened my bags. I wouldn't expect any CPU usage at all from ArkInventory if I'm not even accessing my bags.

youre still looting stuff - that has to be processed. which triggers tooltip count data regen (in the background)

make sure that sorting is set to either manual or on window open.  if its set to instant i think it will recalc the window even if its closed, on open will only do it on open and manual wont do it at all unless you tell it to.

manual is probably the better option (it uses way less everything) if youre not actually fussed about items not being in their correct spot until youre ready for them to be there.

**Comment by ciphersimian** [comment]
In the second example above I deleted my entire WTF folder, so that was with whatever the default is.

As another data point for comparison:
https://imgur.com/a/qrTUrKM

**Comment by arkayenro** [comment]
the default is instant, so youd need to change it

**Comment by ciphersimian** [comment]
It took me forever to find the setting I think you're referring to but I believe it defaults to "On Window Open":

Settings -> Designs (Styles / Layouts) -> Window -> Style -> Sorting -> When

https://imgur.com/a/4x7Bcxx

**Comment by arkayenro** [comment]
if youre getting a spike every 5 seconds does that show up in the addon usage?  ie is there a corresponding cpu spike from an addon?  disable AI and see if the hiccups still happen.

mine uses 130mb of memory but then ive probably got more characters (the more data there is the more memory it needs to store it)

cpu usage is all that really matters, and only if it impacts gameplay

ie i can get 34 ms/s just by mousing over an item - because the tooltip is constantly being rebuilt, the game doesnt hiccup or lag while im doing that though.

**Comment by ciphersimian** [comment]
I don't see any obvious correlation like that but the problems are definitely far more pronounced with the CPU profiling enabled.  I know I have some other addons that are misbehaving (e.g. TSM, Bartender4, Rarity) so I'm not trying to blame the entirety of my performance issues on AI.  I'm just pointing out that it's the tip-top highest CPU usage addon of all over an extended test and that seems insane to me, especially when over the same test a similar addon (AdiBags) is near the bottom.

**Comment by arkayenro** [comment]
> but the problems are definitely far more pronounced with the CPU profiling enabled
i would expect that, enabling profiling has a cpu impact

practically everything in AI is threaded, ie its deliberately slowed down to ensure gameplay isnt affected, it also means that cpu usage goes up because im using it for longer (instead of a massive burst that lags/hiccups the game)
